public/app/plugins/datasource/loki/components/useLokiForceRefresh.ts

```diff
@@ -0,0 +1,27 @@
+import { useState, useEffect } from 'react';
+import { DatasourceStatus } from '@grafana/ui/src/types/plugin';
+
+export const useLokiForceRefresh = (
+  datasourceStatus: DatasourceStatus,
+  refreshLabels: (forceRefresh?: boolean) => void,
+  initialDatasourceStatus?: DatasourceStatus
+) => {
+  const [prevDatasourceStatus, setPrevDatasourceStatus] = useState(
+    initialDatasourceStatus || DatasourceStatus.Connected
+  );
+  const [forceRefresh, setForceRefresh] = useState(false);
+
+  // Effects
+  useEffect(() => {
```

Comment:
You actually don't need that effect at all. The effect below should be sufficient when refactored to:

```ts
useEffect(() => {
    if (datasourceStatus === DatasourceStatus.Connected) {
      refreshLabels(true);
    }
  }, [datasourceStatus]);
```

Since this effect will be perfoemed on `datasourceStatus` change only, then you are secured that the forced refresh will happen when the datasource becomes connected.

Reply:
It will refresh for the initial state then, right?